components/org.wso2.identity.event.http.publisher/src/main/java/org/wso2/identity/event/http/publisher/internal/component/ClientManager.java

```diff
@@ -0,0 +1,229 @@
+            int maxConnectionsPerRoute =
+                    HTTPAdapterDataHolder.getInstance().getAdapterConfiguration().getDefaultMaxConnectionsPerRoute();
+
+            ConnectingIOReactor ioReactor = new DefaultConnectingIOReactor();
+            PoolingNHttpClientConnectionManager asyncConnectionManager =
+                    new PoolingNHttpClientConnectionManager(ioReactor);
+            asyncConnectionManager.setMaxTotal(maxConnections);
+            asyncConnectionManager.setDefaultMaxPerRoute(maxConnectionsPerRoute);
+            LOG.debug("PoolingNHttpClientConnectionManager created with maxConnections: " + maxConnections +
+                    " and maxConnectionsPerRoute: " + maxConnectionsPerRoute);
+            RequestConfig config = createRequestConfig();
+
+            // Initialize HttpAsyncClient
+            HttpAsyncClientBuilder httpAsyncClientBuilder = HttpAsyncClients.custom()
+                    .setDefaultRequestConfig(config)
+                    .setConnectionManager(asyncConnectionManager)
+                    .setSSLContext(createSSLContext());
+            httpAsyncClient = httpAsyncClientBuilder.build();
+            httpAsyncClient.start();
+            LOG.debug("HttpAsyncClient started with config: connectTimeout=" +
+                    config.getConnectTimeout() + ", connectionRequestTimeout=" +
+                    config.getConnectionRequestTimeout() + ", socketTimeout=" +
+                    config.getSocketTimeout() + ", maxConnections=" +
+                    asyncConnectionManager.getMaxTotal() + ", maxConnectionsPerRoute=" +
+                    asyncConnectionManager.getDefaultMaxPerRoute());
+        } catch (IOException e) {
+            throw HTTPAdapterUtil.handleServerException(
+                    ERROR_GETTING_ASYNC_CLIENT, e);
+        }
+    }
+
+    public CloseableHttpAsyncClient getHttpAsyncClient() {
+
+        if (!httpAsyncClient.isRunning()) {
+            LOG.debug("HttpAsyncClient is not running, starting client");
+            httpAsyncClient.start();
+        }
+        return httpAsyncClient;
+    }
+
+    private RequestConfig createRequestConfig() {
+
+        return RequestConfig.custom()
+                .setConnectTimeout(HTTPAdapterDataHolder.getInstance().getAdapterConfiguration()
+                        .getHTTPConnectionTimeout())
+                .setConnectionRequestTimeout(HTTPAdapterDataHolder.getInstance().getAdapterConfiguration()
+                        .getHttpConnectionRequestTimeout())
+                .setSocketTimeout(
+                        HTTPAdapterDataHolder.getInstance().getAdapterConfiguration().getHttpReadTimeout())
+                .setRedirectsEnabled(false)
+                .setRelativeRedirectsAllowed(false)
+                .build();
+    }
+
+    private SSLContext createSSLContext() throws HTTPAdapterException {
+
+        try {
+            return SSLContexts.custom()
+                    .build();
```

Comment:
take to one line

Reply:
fixed with https://github.com/wso2-extensions/identity-event-publishers/pull/37/commits/15538dc2b9ce588c47e8db9c5db39156de457b3c